### PR TITLE
Update for 2018 edition

### PR DESCRIPTION
**Pull request by Goirad**
Makes all the changes for 2018 edition

### CODE REVIEW

**Comment by Goirad** [comment]
I'm realizing now that one of the tests is against compiler version 1.27, how come?

**Comment by steffengy** [comment]
Well, we generally try to stay compatible with the oldest possible version.
Due to dependencies 1.27 is the lowest we can go.

We can go for 1.31 (which is I think all that is required for edition 2018).
I'll probably delay a crates.io release for a while though.

**Comment by Goirad** [comment]
>  I'll probably delay a crates.io release for a while though.
In general, or if we go for 1.31?

**Comment by steffengy** [comment]
In general, but a bit more than usual with 1.31.
Not more than 2 weeks, just giving people a tidbit of time to notice ;)
But you might want to get further changes in anyways in that time window?

**Comment by Goirad** [comment]
Yea that's fair, I probably will have to do more work for [this](https://github.com/sfackler/rust-native-tls/issues/132) anyways.

File: src/key_handle.rs [hunk 1]
```diff
@@ -1,5 +1,3 @@
 //! Deprecated.
-#![deprecated(note = "use cert_context::PrivateKey", since = "0.1.5")]
-
 #[deprecated(note = "use cert_context::PrivateKey", since = "0.1.5")]
-pub use cert_context::PrivateKey as KeyHandle;
+pub use crate::cert_context::PrivateKey as KeyHandlendle;
```
Comment (steffengy) [inline src/key_handle.rs]: `KeyHandlendle` looks like a typo

Comment (Goirad) [reply src/key_handle.rs]: Fixed, but that is bizarre, since that would have been `cargo fix`

Comment (steffengy) [reply src/key_handle.rs]: Usually when I have that it's IDE/editor lag, but yeah very strange ^^

**Comment by steffengy** [comment]
Thanks!

**Pull request merged**